Automattic/wp-calypso
# Plans: Add free trials without going through checkout

This PR updates the components on `/plans` such that a free trial is added to the user's account when they click 'Start Free Trial' without going through checkout.

**Testing**
- [x] *Free trials on `/plans`*
- Run `localStorage.setItem( 'ABTests', '{ "freeTrials_20160120":"offered" }' );`
- Visit `/plans/:site` for a site that has not had a free trial before.
- Click 'Start Free Trial' under one of the plans.
- Assert that the buttons to select a plan are disabled.
- Assert that a 'Submitting' notice appears.
- Assert that you are redirected to `/plans/:site/thank-you`.

- [x] *Plans on `/plans` with free trials enabled*
- Run `localStorage.setItem( 'ABTests', '{ "freeTrials_20160120":"offered" }' );`
- Visit `/plans/:site` for a site that has not had a free trial before.
- Assert that you can add a full plan to your cart by clicking the 'upgrade now' link under the 'Start Free Trial' buttons.
- Assert that you can complete checkout with this plan successfully.

- [x] *Plans on `/plans` with free trials disabled*
- Run `localStorage.setItem( 'ABTests', '{ "freeTrials_20160120":"notOffered" }' );`
- Visit `/plans/:site`.
- Assert that you can complete checkout after adding a plan to your cart by clicking 'Upgrade Now'.

- [x] *Free plan in signup with free trials enabled*
- Run `localStorage.setItem( 'ABTests', '{ "freeTrials_20160120":"offered" }' );`
- Visit `/start`
- Proceed through signup, selecting the free plan.
- Assert that you are redirected to `[site].wordpress.com` after the signup is complete.

- [x] *Paid plans in signup with free trials enabled*
- Run `localStorage.setItem( 'ABTests', '{ "freeTrials_20160120":"offered" }' );`
- Visit `/start`
- Proceed through signup, selecting a paid plan.
- Assert that you are redirected to checkout after signup with the plan you selected in your cart.

- [x] *Free plan in signup with free trials disabled*
- Run `localStorage.setItem( 'ABTests', '{ "freeTrials_20160120":"notOffered" }' );`
- Visit `/start`
- Proceed through signup, selecting the free plan.
- Assert that you are redirected to `[site].wordpress.com` after the signup is complete.

- [x] *Paid plans in signup with free trials disabled*
- Run `localStorage.setItem( 'ABTests', '{ "freeTrials_20160120":"notOffered" }' );`
- Visit `/start`
- Proceed through signup, selecting a paid plan.
- Assert that you are redirected to checkout after signup with the plan you selected in your cart.
